File: app/src/main/java/edu/schoolapp/TodoFragment.java
package edu.schoolapp;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.ListView;

import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.QueryDocumentSnapshot;
import com.google.firebase.firestore.QuerySnapshot;

import java.util.ArrayList;
import java.util.List;

import edu.schoolapp.Model.PrimaryMenuDao;
import edu.schoolapp.Model.TodoModel;

public class TodoFragment extends AppCompatActivity {

    private TodoAdapter adapter;
    private List<TodoModel> todoList;
    private FirebaseFirestore db;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.fragment_todo);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        db = FirebaseFirestore.getInstance();

        ListView todoListView = findViewById(R.id.todo_list_view);
        adapter = new TodoAdapter(todoList);
        todoListView.setAdapter(adapter);

//        getTodoList();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_add, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int selectedItemId = item.getItemId();
        if (selectedItemId == android.R.id.home) {
            finish();
        } else if (selectedItemId == R.id.menu_add) {
            startActivity(new Intent(this, ActivityAddTodo.class));
        }
        return true;
    }

    @Override
    protected void onResume() {
        super.onResume();
        getTodoList();
    }

    private void getTodoList() {
        db.collection("users").document(SchoolApp.getToken()).collection("primaryMenus").document("todo").collection("todolist").get().addOnSuccessListener(new OnSuccessListener<QuerySnapshot>() {
            @Override
            public void onSuccess(QuerySnapshot queryDocumentSnapshots) {
                todoList = new ArrayList<>();
                if (!queryDocumentSnapshots.isEmpty()) {
                    for (QueryDocumentSnapshot snapshots : queryDocumentSnapshots) {
                        TodoModel data = snapshots.toObject(TodoModel.class);
                        todoList.add(data);
                    }
                    adapter.updateData(todoList);
                }
            }
        });
    }
}

File: app/src/main/java/edu/schoolapp/TodoFragment.kt
package edu.schoolapp

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.firestore.FirebaseFirestore
import edu.schoolapp.model.TodoModel

class TodoFragment : AppCompatActivity() {
    private var adapter: TodoAdapter? = null
    private var todoList: MutableList<TodoModel>? = null
    private lateinit var db: FirebaseFirestore
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.fragment_todo)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        db = FirebaseFirestore.getInstance()
        val todoListView = findViewById<ListView>(R.id.todo_list_view)
        adapter = TodoAdapter(todoList)
        todoListView.adapter = adapter

//        getTodoList();
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_add, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val selectedItemId = item.itemId
        if (selectedItemId == android.R.id.home) {
            finish()
        } else if (selectedItemId == R.id.menu_add) {
            startActivity(Intent(this, ActivityAddTodo::class.java))
        }
        return true
    }

    override fun onResume() {
        super.onResume()
        getTodoList()
    }

    private fun getTodoList() {
        SchoolApp.token?.let {
            db.collection("users").document(it).collection("primaryMenus")
                .document("todo").collection("todolist").get()
                .addOnSuccessListener { queryDocumentSnapshots ->
                    todoList = mutableListOf()
                    if (!queryDocumentSnapshots.isEmpty) {
                        for (snapshots in queryDocumentSnapshots) {
                            val data = snapshots.toObject(TodoModel::class.java)
                            todoList?.add(data)
                        }
                        adapter!!.updateData(todoList)
                    }
                }
        }
    }
}